components/CheckBox.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { Checkbox, Collapse } from "antd";
+import "antd/dist/antd.css";
+
+const { Panel } = Collapse;
+
+function CheckBox({ filters, handleFilters }) {
+  let A: {
+    map: any;
+    id: number;
+    name: string;
+    groupId: number;
+    semester: number;
+  }[] = [];
+  const B = [
+    "1st Year - 1st Semester",
+    "1st Year - 2nd Semester",
+    "2nd Year - 1st Semester",
+    "2nd Year - 2nd Semester",
+    "3rd Year - 1st Semester",
+    "3rd Year - 2nd Semester",
+    "4th Year - 1st Semester",
+    "4st Year - 2nd Semester",
+    "5th Year ",
+    "Other",
+  ];
+
+  A[0] = filters.filter((f) => f.groupId === 1 && f.semester === 1);
+  A[1] = filters.filter((f) => f.groupId === 1 && f.semester === 2);
+  A[2] = filters.filter((f) => f.groupId === 2 && f.semester === 1);
+  A[3] = filters.filter((f) => f.groupId === 2 && f.semester === 2);
+  A[4] = filters.filter((f) => f.groupId === 3 && f.semester === 1);
+  A[5] = filters.filter((f) => f.groupId === 3 && f.semester === 2);
+  A[6] = filters.filter((f) => f.groupId === 4 && f.semester === 1);
+  A[7] = filters.filter((f) => f.groupId === 4 && f.semester === 2);
+  A[8] = filters.filter((f) => f.groupId === 5);
+  A[9] = filters.filter((f) => f.groupId === 0);
+
+  let Checked: number[] = [];
+  const handleToggle = (value: number) => {
+    const currentId = Checked.indexOf(value);
+    const newCheck = [...Checked];
+
+    if (currentId === -1) {
+      newCheck.push(value);
+    } else {
+      newCheck.splice(currentId, 1);
+    }
+    handleFilters(newCheck);
+  };
+
+  return (
+    <Collapse>
+      <Panel header="Filtros" key={0}>
+        <Collapse>
+          {B.map((b, index1) => (
+            <Panel header={b} key={index1}>
+              {A[index1]?.map(
+                (
+                  value: {
+                    id: number;
+                    name: string;
+                    groupId: number;
+                    semester: number;
+                  },
+                  index: number
+                ) => (
+                  <React.Fragment key={index}>
+                    <div>
+                      <Checkbox
+                        onChange={() => handleToggle(value.id)}
```

Comment:
The box it's not being properly checked, so that, I can't uncheck it. Could you fix this? 🙏🏼 

![image](https://user-images.githubusercontent.com/76881129/189525079-c3c0f5be-470b-425d-afe7-ae5e2b97d2bd.png)